react-components/src/analysis/view/AnalysesView.js

```diff
@@ -538,7 +552,8 @@ class AnalysesView extends Component {
     }
 
     handleSelectAllClick(event, checked) {
-        if (checked) {
+        const { selected } = this.state;
+        if (checked && !selected.length) {
```

Comment:
I added this check so that everything will be selected only if the selection was already empty.

When only some items in the listing are selected, the select-all checkbox icon becomes the indeterminate icon, which looks like a minus sign to me ➖ . Clicking on this will now remove all selections, rather than selecting everything. This is also how Gmail behaves.